### PR TITLE
Introduce Preliminary Documentation for Storage

### PR DESCRIPTION
**Pull request by chirangaalwis**

### CODE REVIEW

File: store/Persisting_And_Sharing.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  - WSO2 product Helm charts use the stable [NFS Server Provisioner](https://hub.helm.sh/charts/stable/nfs-server-provisioner)
+    to render a NFS server, for the purpose of evaluation.
+    
+  - WSO2 product deployments have been tried and tested with NFS based solutions like [Google FileStore](https://cloud.google.com/filestore)
+    (please refer to instructions for [API Manager](https://github.com/wso2/kubernetes-apim/issues/359#issuecomment-639729986) and
+    [Identity Server](https://github.com/wso2/kubernetes-is/issues/227#issuecomment-639735208)).
+
+* Ceph File System (CephFS)
+
+  - WSO2 product deployments have been tested with [CephFS cluster deployed using the Rook Operator](https://rook.io/docs/rook/v1.3/ceph-quickstart.html).
```
Comment (ThilinaManamgoda) [inline store/Persisting_And_Sharing.md]: Would it be better if we mention the Rook Operator version here even though the link is directed to the 1.3. docs.

Comment (chirangaalwis) [reply store/Persisting_And_Sharing.md]: +1

**Review by ThilinaManamgoda — APPROVED** [review]

**Review by DilanUA — APPROVED** [review]

File: store/Persisting_And_Sharing.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+## Contents
+
+The following will be discussed in detail in the document.
+
+* [4.1. Why do we need to store?](#why-do-we-need-to-store?)
+* [4.2. Recommended storage options for WSO2 products](#recommended-storage-options-for-wso2-products)
+
+### Why do we need to store?
+
+In the context of a WSO2 product deployment, the need for storage of artifacts arises due to the following reasons.
```
Comment (chamindi-a) [inline store/Persisting_And_Sharing.md]:
```suggestion
In the context of WSO2 product deployment, the need for storage of artifacts arises due to the following reasons:
```

File: store/Persisting_And_Sharing.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+* Persist created runtime artifacts
+
+* Share created runtime artifacts across multiple instances in deployments with high availability support
+
+> For example, in API Manager production deployment patterns, it is recommended to use a suitable storage option
```
Comment (chamindi-a) [inline store/Persisting_And_Sharing.md]:
```suggestion
> For example, in WSO2 API Manager production deployment patterns, it is recommended to use a suitable storage option
```

File: store/Persisting_And_Sharing.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  for the aforementioned purposes.
+>
+> Following are some such scenarios.
+>
+>   - Persist Apache Solr-based indexed data in every Publisher and DevPortal profile instance, individually
+>
+>   - Persist and share API runtime artifacts (created in `<APIM_HOME>/repository/deployment/server/synapse-configs`) directory
+>     across multiple instances of the API Manager Gateway profile
+>
+> For advanced details on persistent runtime artifacts of API Manager, please refer to the official
```
Comment (chamindi-a) [inline store/Persisting_And_Sharing.md]:
```suggestion
> For advanced details on persistent runtime artifacts of WSO2 API Manager, please refer to the official
```

File: store/Persisting_And_Sharing.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+[documentation](https://apim.docs.wso2.com/en/latest/install-and-setup/setup/reference/common-runtime-and-configuration-artifacts/#persistent-runtime-artifacts).
+
+### Recommended storage options for WSO2 products
+
+As highlighted in the [previous](#why-do-we-need-to-store?) section, there are numerous use cases which instigate the need
+for storage in WSO2 product deployments.
+
+WSO2 recommends using the most suitable persistent storage solution depending on the scenario.
+
+* Persist runtime artifacts of a single instance, only
```
Comment (chamindi-a) [inline store/Persisting_And_Sharing.md]:
```suggestion
* Persist runtime artifacts of a single instance: 
```

File: store/Persisting_And_Sharing.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+  For this purpose, you may use any persistent storage solution supporting either `ReadWriteOnce` or `ReadWriteMany`
+  [access modes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes).
+
+* Persist and share runtime artifacts runtime artifacts across multiple instances of a product profile
```
Comment (chamindi-a) [inline store/Persisting_And_Sharing.md]:
```suggestion
* Persist and share runtime artifacts across multiple instances of a product profile
```

File: store/Persisting_And_Sharing.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+  For this purpose, you may use any persistent storage solution supporting `ReadWriteMany`
+  [access mode](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes).
+
+The following is a list of storage options using with production grade WSO2 product deployments in Kubernetes environments
```
Comment (chamindi-a) [inline store/Persisting_And_Sharing.md]:
```suggestion
The following is a list of storage options using with production grade WSO2 product deployments in Kubernetes environments that
```

File: store/Persisting_And_Sharing.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+have been tested with along with corresponding [Kubernetes StorageClasses](https://kubernetes.io/docs/concepts/storage/storage-classes/).
```
Comment (chamindi-a) [inline store/Persisting_And_Sharing.md]:
```suggestion
have been tested along with the corresponding [Kubernetes StorageClasses](https://kubernetes.io/docs/concepts/storage/storage-classes/).
```

File: store/Persisting_And_Sharing.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+* Network File System (NFS)
+
+  - WSO2 product Helm charts use the stable [NFS Server Provisioner](https://hub.helm.sh/charts/stable/nfs-server-provisioner)
+    to render a NFS server, for the purpose of evaluation.
+    
+  - WSO2 product deployments have been tried and tested with NFS based solutions like [Google FileStore](https://cloud.google.com/filestore)
+    (please refer to instructions for [API Manager](https://github.com/wso2/kubernetes-apim/issues/359#issuecomment-639729986) and
```
Comment (chamindi-a) [inline store/Persisting_And_Sharing.md]:
```suggestion
    (please refer to instructions for [WSO2 API Manager](https://github.com/wso2/kubernetes-apim/issues/359#issuecomment-639729986) and
```

File: store/Persisting_And_Sharing.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+    [Identity Server](https://github.com/wso2/kubernetes-is/issues/227#issuecomment-639735208)).
```
Comment (chamindi-a) [inline store/Persisting_And_Sharing.md]:
```suggestion
    [WSO2 Identity Server](https://github.com/wso2/kubernetes-is/issues/227#issuecomment-639735208)).
```

File: store/Persisting_And_Sharing.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+* Ceph File System (CephFS)
+
+  - WSO2 product deployments have been tested with [CephFS cluster deployed using the Rook Operator version `1.3`](https://rook.io/docs/rook/v1.3/ceph-quickstart.html).
+    (please refer to instructions for [API Manager](https://github.com/wso2/kubernetes-apim/issues/410#issuecomment-654070688) and
```
Comment (chamindi-a) [inline store/Persisting_And_Sharing.md]:
```suggestion
    (Refer to instructions for [WSO2 API Manager](https://github.com/wso2/kubernetes-apim/issues/410#issuecomment-654070688) and
```

File: store/Persisting_And_Sharing.md [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+    [Identity Server](https://github.com/wso2/kubernetes-is/issues/240#issuecomment-654681300)).
```
Comment (chamindi-a) [inline store/Persisting_And_Sharing.md]:
```suggestion
    [WSO2 Identity Server](https://github.com/wso2/kubernetes-is/issues/240#issuecomment-654681300)).
```

**Review by chamindi-a — CHANGES_REQUESTED** [review]
@chirangaalwis: Make sure the anchor tags are properly working in the read me 
and store/Persisting_And_Sharing.md is been looking at

File: store/Persisting_And_Sharing.md (hunk 2 above)
Comment (chirangaalwis) [reply store/Persisting_And_Sharing.md]: +1

File: store/Persisting_And_Sharing.md (hunk 6 above)
Comment (chirangaalwis) [reply store/Persisting_And_Sharing.md]: @chamindi-a +1 but there is an extra space here. Can you send a new suggestion for this?

Comment (chamindi-a) [reply store/Persisting_And_Sharing.md]: @chirangaalwis: done

**Review by chamindi-a — APPROVED** [review]

**Pull request merged**